src/js/api/posts/create.mjs

```diff
@@ -0,0 +1,30 @@
+// Functionality to create a post
+
+import { authFetch } from "../authFetch.mjs";
+import { apiSocialUrl } from "../constants.mjs";
+
+const action = "/posts";
+const method = "post";
+
+export async function createPost(postData) {
+    const createPostUrl = apiSocialUrl + action;
+
+    try {
+        const response = await authFetch(createPostUrl, {
+            method,
+            body: JSON.stringify(postData)
+        });
+        if (response.ok) {
+            location.reload();
+            return await response.json ();
+        } else {
+            throw new Error ("Response not ok");
```

Comment:
If the error is 400, there will be a helpful message explaining what went wrong. It's worth passing this to the user.